### PR TITLE
Expose a few more OpenSSL functions that are useful for DTLS support

### PR DESCRIPTION
**Pull request by njsmith**
Also `OP_NO_RENEGOTIATION` I guess, which is useful because renegotiation is the worst.

### CODE REVIEW

File: src/_cffi_src/openssl/ssl.py [hunk 1]
```diff
@@ -468,6 +476,13 @@
 long DTLSv1_handle_timeout(SSL *);
 long DTLS_set_link_mtu(SSL *, long);
 long DTLS_get_link_min_mtu(SSL *);
+size_t DTLS_get_data_mtu(SSL *);
+long SSL_set_mtu(SSL *, long);
+typedef ... BIO_ADDR;
```
Comment (reaperhulk) [inline src/_cffi_src/openssl/ssl.py:481]: This should be in types

Comment (reaperhulk) [reply src/_cffi_src/openssl/ssl.py:481]: Where are these BIO_ADDR things originally defined? If bio.h they should live in bio.py

Comment (njsmith) [reply src/_cffi_src/openssl/ssl.py]: Fixed

**Comment by njsmith** [comment]
...This thing where each CI run has to be re-approved again is pretty annoying. I guess feel free to add me to the org temporarily? I promise that the only way I will abuse my power would be to kick myself out again ASAP.

**Comment by alex** [comment]
An easier move, more compatible with how we manage permissions generally, would be if you could send some other small PR (docs fix, typo, etc.) and then we could merge that, freeing this up to be run CI jobs without our intervention.

**Comment by njsmith** [comment]
Well, I guess I'll keep an eye out for typos then? :-) In the mean time, I *think* it should pass now, so probably it will only take another 3 attempts...

**Comment by njsmith** [comment]
Well would you look at that. I guess this is ready for review then.

File: src/_cffi_src/openssl/ssl.py [hunk 2]
```diff
@@ -594,6 +614,10 @@
 long (*DTLS_get_link_min_mtu)(SSL *) = NULL;
 #endif
 
+#if CRYPTOGRAPHY_OPENSSL_LESS_THAN_111
+size_t (*DTLS_get_data_mtu)(SSL *) = NULL;
```
Comment (reaperhulk) [inline src/_cffi_src/openssl/ssl.py:618]: Take a look in `_conditional.py` to see what we do when a symbol isn't actually available.

File: src/_cffi_src/openssl/ssl.py [hunk 3]
```diff
@@ -556,6 +569,13 @@
 static const long Cryptography_HAS_NEXTPROTONEG = 0;
 static const long Cryptography_HAS_ALPN = 1;
 
+#ifdef SSL_OP_NO_RENEGOTIATION
+static const long Cryptography_HAS_OP_NO_RENEGOTIATION = 1;
+#else
+static const long Cryptography_HAS_OP_NO_RENEGOTIATION = 0;
```
Comment (reaperhulk) [inline src/_cffi_src/openssl/ssl.py:575]: This should be used in `_conditional.py` to remove the value when it doesn't exist.

File: src/_cffi_src/openssl/ssl.py (hunk 2 above)
Comment (njsmith) [reply src/_cffi_src/openssl/ssl.py:618]: It looks like currently, `_conditional.py` doesn't actually remove any conditionally-available functions (e.g. `DTLS_get_link_min_mtu`, visible just above my new code in the diff, doesn't appear there). So I made my best guess at how it should work for `DTLS_get_data_mtu` -- take a look.

Comment (reaperhulk) [reply src/_cffi_src/openssl/ssl.py:618]: Ha, I would argue that is a bug (we don't want callables that are null func ptrs!). Looks like that's a thing we need to separately fix for the DTLS bindings though...

**Comment by alex** [comment]
Looks like there's some flake8 issues.

**Comment by njsmith** [comment]
Huh. Codecov output is a bit mangled, but I *think* it might be saying that in `_comparisons.py`, we're never detecting `DTLS_get_data_mtu` as being falsey? That doesn't make sense to me -- we're definitely setting it to NULL on libressl and openssl 1.1.0, and I confirmed locally that CFFI treats null function pointers as falsey (`bool(ffi.cast("void (*)(void)", 0)) == False`).

**Comment by njsmith** [comment]
image

**Pull request merged**